mock/src/components/LoginButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { Dispatch, SetStateAction } from "react";
+import { useState } from "react";
+
+/**
+ * This component deals with the "Login" button interactions on click
+ */
+
+interface loginProps {
+  isLoggedIn: boolean;
+  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
+}
+
+export function LoginButton(props: loginProps) {
+  /**
+   * A function then returns the updated value of newValue, which indicates the
+   * new state of the isLoggedIn variable after the toggle
+   *
+   * When called, it flips the authentication state and updates the
+   * corresponding state in the parent component.
+   *
+   * @returns a boolean - indicates if isLoggedIn is now true or false after
+   * being called onClick
+   */
+  const [username, setUsername] = useState("");
+
+  const handleUsername = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUsername(e.target.value);
+  };
+
+  const authenticate = () => {
+    if (props.isLoggedIn) {
+      // If already logged in, sign out
+      props.setIsLoggedIn(false);
+      setUsername(""); // Clear the username
+    } else {
+      if (username.endsWith("@brown.edu")) {
+        const newValue = !props.isLoggedIn;
+        props.setIsLoggedIn(newValue);
+        return newValue;
+      } else {
+        alert("Invalid email format. Please use a @brown.edu email.");
```

Comment:
This pop up message was a great feature for usability! Even though I wasn't sure what to input originally, I was able to use the Mock properly after this message was displayed.